src/v/kafka/server/handlers/fetch.cc

```diff
@@ -1418,6 +1424,22 @@ class simple_fetch_planner final : public fetch_planner::impl {
                   return;
               }
 
+              const auto& topic_md = metadata_cache.get_topic_metadata_ref(
```

Comment:
comment on the overall purpose of the code

Reply:
Did so below.